scripts/mvf_copy.py

```diff
@@ -39,13 +39,53 @@
 from katdal.lazy_indexer import dask_getitem
 
 
+DESCRIPTION = """
+Copy MVFv4 dataset from S3/disk to disk using dask.
+
+Run the script like this:
+
+  mvf_copy.py https://archive/1698676533/1698676533_sdp_l0.full.rdb?token=<> dest
+
+or:
+
+  mvf_copy.py src_dir dest_dir
+
+Data will appear in three subdirectories in the specified output directory as
+
+  dest/1698676533/...
+  dest/1698676533-sdp-l0/...
+  dest/1698676533-sdp-l1-flags/...
+
+Open the local dataset like this:
+
+  d = katdal.open("dest/1698676533/1698676533_sdp_l0.full.rdb")
+
+While dask allows multiple retries while downloading chunks, it currently has
+no way to resume copying if the script crashes. For peace of mind, consider
+using the mvf_download.py script instead.
+
+BONUS: you can even copy just parts of the data by selecting a subset of
+correlation products. The --corrprods value is passed to DataSet.select().
+
+Some examples:
+
+  mvf_copy.py url directory --corrprods=auto
+  mvf_copy.py url directory --corrprods=cross
+"""
+
+
 def parse_args():
-    parser = argparse.ArgumentParser()
+    """Parse script arguments."""
+    parser = argparse.ArgumentParser(
+        usage='%(prog)s [-h] [--corrprods CORRPRODS] [--workers N] source dest',
+        description=DESCRIPTION,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
     parser.add_argument('source', help='Dataset URL (or input RDB file path)')
     parser.add_argument('dest', type=Path, help='Output directory')
     parser.add_argument('--corrprods',
-                        help='Select correlation products (kwarg to katdal.DataSet.select). '
-                        'Keeps all corrprods by default.')
+                        help='Select correlation products (kwarg to '
+                             'katdal.DataSet.select). Keeps all corrprods by default.')
     parser.add_argument('--workers', type=int, default=8 * dask.system.CPU_COUNT,
```

Comment:
`8 * dask.system.CPU_COUNT` Although this number of workers is ok for client side of things , the server side of things could get overwhelmed fairly quickly if  multiple users or downloads are done in parallel as would be expected for this script.

Reply:
I challenge you to benchmark this and come up with a better number once and for all 😁